homeassistant/helpers/template.py

```diff
@@ -1262,7 +1262,7 @@ class TemplateEnvironment(ImmutableSandboxedEnvironment):
 
     def __init__(self, hass):
         """Initialise template environment."""
-        super().__init__()
+        super().__init__(loader=jinja2.FileSystemLoader("/config/templates"))
```

Comment:
This should not have a statically configured path, but use the actual configuration path of the system instead. Besides, should this be fixed to a `templates` subfolder?

Reply:
Thanks @frenck .  Very good point.  How does one determine the actual configuration path?

As for the fixed `templates` subfolder, I did that for security reasons to separate out from other config files and ensure one could not, for example, `{% include 'secrets.yaml' %}`

Reply:
Its in the hass object: `hass.config.config_dir`.

I kinda get the security concern, however, if one has access to this level of things, that might be a false sense of security already. The downside of limiting/locking it to the `templates` folder, is limiting the user to structure the configuration to their own likings (e.g., packages use).

Reply:
Ah, thanks.  Didn't see `config_dir` in the docs!

Well, there is nothing stopping the user from creating a `/config/templates/{X,Y,Z}/...` structure for packages.

Reply:
Unfortunately, @frenck, that doesn't work:
```txt
homeassistant/helpers/template.py:1265: in __init__
    super().__init__(loader=jinja2.FileSystemLoader(hass.config.config_dir+"templates"))
E   AttributeError: 'NoneType' object has no attribute 'config'
```
Apparently `__init__` is being called with hass=None at some point.  I don't know if there is a way to lazy-eval that?

Reply:
I think I know how to solve this...  But it means that the feature will only work if there is a hass env.

Reply:
@frenck -- I have this now working locally based on `config_path`, but the code is a little ugly because I can only instantiate the loader if hass is not None, so I have to call `super.__init__()` differently.  I can push now if you want to see it.

Now to add tests as per your other request; how can I add a "file" in the config path in the test infrastructure?  I can do the "stupid" think and literally open/write/close/unlink -- but is there a better way to push something into the filesystem under the test?

Reply:
I now believe I have made the config and test changes you requested.